medcat/config.py

```diff
@@ -111,204 +199,297 @@ def load(cls, save_path: str) -> "ConfigMixin":
-                # If True it will filter before doing disamb. Useful for the trainer.
-                'filter_before_disamb': False,
-                # Concepts that have seen less training examples than this will not be used for
-                #similarity calculation and will have a similarity of -1.
-                'train_count_threshold': 1,
-                # Do we want to calculate context similarity even for concepts that are not ambigous.
-                'always_calculate_similarity': False,
-                # Weights for a weighted average
-                #'weighted_average_function': partial(weighted_average, factor=0.02),
-                'weighted_average_function': partial(weighted_average, factor=0.0004),
-                # Concepts below this similarity will be ignored. Type can be static/dynamic - if dynamic each CUI has a different TH
-                #and it is calcualted as the average confidence for that CUI * similarity_threshold. Take care that dynamic works only
-                #if the cdb was trained with calculate_dynamic_threshold = True.
-                'calculate_dynamic_threshold': False,
-                'similarity_threshold_type': 'static',
-                'similarity_threshold': 0.25,
-                # Probability for the negative context to be added for each positive addition
-                'negative_probability': 0.5,
-                # Do we ignore punct/num when negative sampling
-                'negative_ignore_punct_and_num': True,
-                # If >0 concepts for which a detection is its primary name will be preferred by that amount (0 to 1)
-                'prefer_primary_name': 0.35,
-                # If >0 concepts that are more frequent will be prefered by a multiply of this amount
-                'prefer_frequent_concepts': 0.35,
-                # DISABLED in code permanetly: Subsample during unsupervised training if a concept has received more than
-                'subsample_after': 30000,
-                # When adding a positive example, should it also be treated as Negative for concepts
-                #which link to the postive one via names (ambigous names).
-                'devalue_linked_concepts': False,
-                # If true when the context of a concept is calculated (embedding) the words making that concept are not taken into accout
-                'context_ignore_center_tokens': False,
-                # Filters
-                'filters': {
-                    'cuis': set(), # CUIs in this filter will be included, everything else excluded, must be a set, if empty all cuis will be included
-                    },
-                }
-
-
-        # Some regex that we will need
-        self.word_skipper = re.compile('^({})$'.format('|'.join(self.preprocessing['words_to_skip'])))
-        # Very agressive punct checker, input will be lowercased
-        self.punct_checker = re.compile(r'[^a-z0-9]+')
+    def asdict(self) -> Dict[str, Any]:
```

Comment:
Not sure if this or the above change would be needed if FakeDict or MixingConfig directly inherit from `dict` or `UserDict`